docs/iris/src/whatsnew/2.3.rst

```diff
@@ -0,0 +1,193 @@
+What's New in Iris 2.3
+**********************
+
+:Release: 2.3
```

Comment:
@trexfeathers Can we explicitly clarify this as `2.3.0`

Reply:
👍